armotypes/posturetypes.go

```diff
@@ -138,6 +138,9 @@ type PostureClusterSummary struct {
 	FullName   string `json:"clusterFullName"`
 	ShortName  string `json:"clusterShortName"`
 	PrefixName string `json:"clusterPrefixName"`
+
+	//tickets opened for in this cluster
+	Tickets []Ticket `json:"tickets,omitempty"`
```

Comment:
What about generics?
like have 
```golang

type SingleObjectResponseGeneric[T any] struct {
	Value T
}

type ResponseGen[T any] struct {
	SingleObjectResponseGeneric[T]
	Tickets []string `json:"tickets"`
}

type ResourceWithTickets armotypes.V2ListResponseGeneric[[]ResponseGen[armotypes.PostureResource]]
```
The ResourceWithTickets can be defined only in the BL layer

Reply:
This will save us the need to have new armotypes release just to add tickets to our APIs

Reply:
@Bezbran 
What is the benefit ?

Reply:
I wrote. no need to change all the structures

Reply:
now and future

Reply:
OK, still we should wrap the real business data unit with the tickets, which are part of system wide enrichment.
So you may have the ResourceWithTickets in the armotypes, but we shouldn't change each and every struct for this.

Reply:
I still don't see why we need to create a type for every field we want to enrich on few structs 
We don't have a type f0r structs with cluster short name for instance 
I would agree if this was a conditional field - but here it is part of the model

Reply:
Because it's none of the business of the resource that there are ticketes realted to it...

Reply:
I don't like generic response types 
If we want to split the types logic we need to do it with explicit type and composition - not generic with responses that create on the fly types 

e.g.
```go

type DashboradCluster struct{
     PostureCluster
     Tickets []Tickest
     SmartRemidiation 
     IgnoreRuleSummary ...

} 
```

Reply:
@Bezbran 
Other option with generics that still keeps an explicit type (BLCluster)
It can be a good idea but I don't see how it is related to the scope of this task 
```go
type BLEnrichedObject[T any] struct{
   Tickets []Tickets
   SmartRemidiation SmartRemidiation 
}

type BLCluster BLEnrichedObject[Cluster]
```

Reply:
let's not do only only for tickets, it will make it harder to understand the concept and the flows

Reply:
I meant to something like this BLEnrichedObject

Reply:
again, if each task will change the fundamental structs, we will end up with a mess

Reply:
I think I cleared my view, if you want to go ahead go ahead